src/main/java/org/veupathdb/service/eda/merge/core/request/MergedTabularRequestResources.java

```diff
@@ -51,6 +63,71 @@ public MergedTabularRequestResources(MergedEntityTabularPostRequest request, Ent
     validateIncomingRequest();
   }
 
+  /**
+   * TODO migrate this to directly use compute functionality:
+   *  org.veupathdb.service.eda.compute.controller.ComputeController#resultFile(
+   *  org.veupathdb.service.eda.compute.plugins.PluginMeta, java.lang.String,
+   *  org.veupathdb.service.eda.generated.model.ComputeRequestBase, java.util.function.Function)
+   *
+   *  This isn't expected to provide a particularly noticable performance improvement, but it will avoid overhead
+   *  of HTTP.
+   */
+  public CloseableIterator<Map<String, String>> getComputeStreamIterator(Study study) {
+    Entity computeEntity = study.getEntity(_computeInfo.get().getComputeEntity()).orElseThrow();
+
+    // Construct headers from metadata.
+    List<String> headers = Stream.concat(Stream.of(computeEntity.getId() + "." + computeEntity.getPKColName()), Stream.concat(
+            computeEntity.getAncestorPkColNames().stream()
+                .map(pk -> computeEntity.getId() + "." + pk),
+            _computeInfo.get().getVariables().stream()
+                .map(var -> VariableDef.toDotNotation(var.getVariableSpec()))
+        ))
+        .toList();
+
+    DelimitedDataParser d = new DelimitedDataParser(headers, TAB, true);
```

Comment:
I might defer this, because this functionality is going away in the near term and I'd fear a bit of wasted effort if the new utility code had 0 usages.